docs/rules/no-restricted-properties.md

```diff
@@ -0,0 +1,73 @@
+# Disallow certain object properties (no-restricted-properties)
+
+Certain properties on objects may be disallowed in a codebase. This is useful for deprecating an API or restricting usage of a module's methods. This rule looks for accessing a given property key on a given object name, either when reading the property's value or invoking it as a function. You may specify an optional message to indicate an alternative API or a reason for the restriction.
+
+## Rule Details
+
+This rule is aimed at disallowing certain object properties from your code.
```

Comment:
We just started replacing “aims at” or “is aimed at” with purposefully un-DRY repetition of the description with additional context. Suggest moving the last 2 sentences from intro paragraph to Rule Details. I added [variable] in brackets as a placeholder to suggest we somehow make it clear that the rule depends on consistent variable naming in the code, not inference of object type, true?

This rule disallows the use of specified object properties from your code. This rule reports references to a specified property key on a specified object [variable] name, either when reading the property's value or invoking it as a function. You may specify an optional message to indicate an alternative API or a reason for the restriction.